ui/src/buckets/components/UpdateBucketOverlay.tsx

```diff
@@ -1,153 +1,159 @@
 import BucketOverlayForm from 'src/buckets/components/BucketOverlayForm'
 
 // Actions
 import {updateBucket} from 'src/buckets/actions'
+import {notify} from 'src/shared/actions/notifications'
+
+// APIs
+import * as api from 'src/client'
 
 // Constants
 import {DEFAULT_SECONDS} from 'src/buckets/components/Retention'
+import {getBucketFailed} from 'src/shared/copy/notifications'
 
-// Types
-import {AppState, Bucket} from 'src/types'
-
-interface State {
-  bucket: Bucket
-  ruleType: 'expire'
-}
-
-interface StateProps {
-  bucket: Bucket
-}
+//Types
+import {Bucket} from 'src/types'
 
 interface DispatchProps {
   onUpdateBucket: typeof updateBucket
+  onNotify: typeof notify
 }
 
-type Props = StateProps & DispatchProps & WithRouterProps
-
-class UpdateBucketOverlay extends PureComponent<Props, State> {
-  constructor(props) {
-    super(props)
-
-    const {bucket} = this.props
-
-    this.state = {
-      ruleType: this.ruleType(bucket),
-      bucket,
-    }
-  }
+type Props = DispatchProps & WithRouterProps
 
-  public render() {
-    const {bucket, ruleType} = this.state
+const UpdateBucketOverlay: FunctionComponent<Props> = ({
+  onUpdateBucket,
+  onNotify,
+  params: {bucketID, orgID},
+  router,
+}) => {
+  const [bucketDraft, setBucketDraft] = useState<Bucket>(null)
 
-    return (
-      <Overlay visible={true}>
-        <Overlay.Container maxWidth={500}>
-          <Overlay.Header title="Edit Bucket" onDismiss={this.handleClose} />
-          <Overlay.Body>
-            <BucketOverlayForm
-              name={bucket.name}
-              buttonText="Save Changes"
-              ruleType={ruleType}
-              onCloseModal={this.handleClose}
-              onSubmit={this.handleSubmit}
-              disableRenaming={true}
-              onChangeInput={this.handleChangeInput}
-              retentionSeconds={this.retentionSeconds}
-              onChangeRuleType={this.handleChangeRuleType}
-              onChangeRetentionRule={this.handleChangeRetentionRule}
-            />
-          </Overlay.Body>
-        </Overlay.Container>
-      </Overlay>
-    )
-  }
+  const [loadingStatus, setLoadingStatus] = useState(RemoteDataState.Loading)
 
-  private get retentionSeconds(): number {
-    const rule = this.state.bucket.retentionRules.find(r => r.type === 'expire')
+  const [retentionSelection, setRetentionSelection] = useState(DEFAULT_SECONDS)
 
-    if (!rule) {
-      return DEFAULT_SECONDS
-    }
+  useEffect(() => {
+    const fetchBucket = async () => {
+      const resp = await api.getBucket({bucketID})
 
-    return rule.everySeconds
-  }
+      if (resp.status !== 200) {
+        onNotify(getBucketFailed(bucketID, resp.data.message))
+        handleClose()
+        return
+      }
+      setBucketDraft(resp.data)
 
-  private ruleType = (bucket: Bucket): 'expire' => {
-    const rule = bucket.retentionRules.find(r => r.type === 'expire')
+      const rules = get(resp.data, 'retentionRules', [])
```

Comment:
it would be cool if this was also in an async function that awaited the bucket data. i know they're super small functions that reuse the same data, they just feel chainable and self contained